pubsub/pulsar/pulsar.go

```diff
@@ -173,9 +173,28 @@ func (p *Pulsar) Init(_ context.Context, metadata pubsub.Metadata) error {
 		ConnectionTimeout:          30 * time.Second,
 		TLSAllowInsecureConnection: !m.EnableTLS,
 	}
-	if m.Token != "" {
+
+	switch {
+	case len(m.Token) > 0:
```

Comment:
I didn't notice that token was optional. I think in this case it makes even more sense to do the auth this way and not having an explicit "authType"?